src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.Unix.cs

```diff
@@ -114,6 +143,136 @@ private TimeZoneInfo(byte[] data, string id, bool dstDisabled)
+            }
+
+            if (genericName == null)
+            {
+                // When we can't get a generic name, use the offset and the ID.
+                // It is not ideal, but at least it is non-ambiguous.
+                // (Note, UTC was handled already above.)
+                displayName = $"{baseOffsetText} {timeZoneId}";
+                return;
+            }
+
+            // Get the generic location name.
+            string? genericLocationName = null;
+            GetDisplayName(timeZoneId, Interop.Globalization.TimeZoneDisplayNameType.GenericLocation, uiCulture.Name, ref genericLocationName);
+
+            // Some edge cases only apply when the offset is +00:00.
+            if (baseUtcOffset == TimeSpan.Zero)
+            {
+                // GMT and its aliases will just use the equivalent of "Greenwich Mean Time".
+                string? gmtLocationName = null;
+                GetDisplayName(GmtId, Interop.Globalization.TimeZoneDisplayNameType.GenericLocation, uiCulture.Name, ref gmtLocationName);
+                if (genericLocationName == gmtLocationName)
+                {
+                    displayName = $"{baseOffsetText} {genericName}";
+                    return;
+                }
+
+                // Other zones with a zero offset and the equivalent of "Greenwich Mean Time" should only use the location name.
+                // For example, prefer "Iceland Time" over "Greenwich Mean Time (Reykjavik)".
+                string? gmtGenericName = null;
+                GetDisplayName(GmtId, Interop.Globalization.TimeZoneDisplayNameType.Generic, uiCulture.Name, ref gmtGenericName);
+                if (genericName == gmtGenericName)
+                {
+                    displayName = $"{baseOffsetText} {genericLocationName}";
+                    return;
+                }
+            }
+
+            if (genericLocationName == genericName)
+            {
+                // When the location name is the same as the generic name,
+                // then it is generally good enough to show by itself.
+
+                // *** Example (en-US) ***
+                // id                   = "America/Havana"
+                // baseOffsetText       = "(UTC-05:00)"
+                // standardName         = "Cuba Standard Time"
+                // genericName          = "Cuba Time"
+                // genericLocationName  = "Cuba Time"
+                // exemplarCityName     = "Havana"
+                // displayName          = "(UTC-05:00) Cuba Time"
+
+                displayName = $"{baseOffsetText} {genericLocationName}";
+                return;
+            }
+
+            // Also use location names in some special cases.  (See the list at the top of this file.)
+            if (ZonesThatUseLocationName.Contains($"\n{timeZoneId}\n", StringComparison.OrdinalIgnoreCase))
```

Comment:
>timeZoneId [](start = 55, length = 10)

This is going to allocate a new string. if we need to optimize that to avoid the allocation, maybe consider providing the custom Contain method like the following:

```C#
        private static bool Contains(string pattern, string source)
        {
            Debug.Assert(source.Length > 1);

            int index = 1;
            do
            {
                index = source.IndexOf(pattern, index, StringComparison.OrdinalIgnoreCase);
                if (index < 0)
                {
                    return false;
                }

                if (index + pattern.Length < source.Length && source[index - 1] == '\n' && source[index + pattern.Length] == '\n')
                {
                    return true;
                }

                index += pattern.Length;
            } while (index < source.Length);

            return false;
        }
```

Reply:
Is this the recommended approach? I really just want to have the affect of searching an array of strings for an exact match, but you had said previously to use a substring search.  I didn't want to risk a future situation where some time zone id is a substring of some other time zone id, so a delimiter on each side is required to do the substring search without risk of collision.

Reply:
I went ahead with this approach. Happy to change it out if there's a simpler way, but I do see the efficiencies here.

Reply:
Why not just a simple array?

```C#
private static readonly string[] ZonesThatUseLocationName = new[] { "One", "Two", "Three" };
```

Reply:
@eerhardt this was my suggestion to use string and just search it instead of looping on array items. I was not expecting we'll need to concatenate the string we are searching for with other bound characters. If this the case, I don't mind reverting back to array usage.

Reply:
What was the original concern to not use a `string[]`? Performance?

Reply:
performance and code simplicity. In addition, the string would be defined as const which wouldn't cause any allocation like the array. 

compare 
```
if (source.Contain(something, Ordinal))
```

to 
```
for (int i = 0; i < array.Length; i++)
{
	if (source.Equals(something, Ordinal))
    {
       break;
    }
}

if (i < array.Length) // found
{
}

```

Reply:
> performance

Doing a quick benchmark:

```C#
    [Benchmark]
    public bool SimpleWay()
    {
        return ContainsSimple("Europe/Minsk", ZonesThatUseLocationNameArray) &&
            (ContainsSimple("America/Los_Angeles", ZonesThatUseLocationNameArray) ||
            ContainsSimple("Europe/Lisbon", ZonesThatUseLocationNameArray) ||
            ContainsSimple("Africa/Nairobi", ZonesThatUseLocationNameArray));
    }

    private static readonly string[] ZonesThatUseLocationNameArray = new[]
    {
        "Europe/Minsk",      // Prefer "Belarus Time" over "Moscow Standard Time (Minsk)"
        "Europe/Moscow",      // Prefer "Moscow Time" over "Moscow Standard Time"
        "Europe/Simferopol", // Prefer "Simferopol Time" over "Moscow Standard Time (Simferopol)"
        "Pacific/Apia",      // Prefer "Samoa Time" over "Apia Time"
        "Pacific/Pitcairn"   // Prefer "Pitcairn Islands Time" over "Pitcairn Time"
    };

    private static bool ContainsSimple(string pattern, string[] source)
    {
        foreach (string s in source)
        {
            if (s == pattern)
            {
                return true;
            }
        }

        return false;
    }


    [Benchmark]
    public bool CurrentWay()
    {
        return ContainsCurrent("Europe/Minsk", ZonesThatUseLocationName) &&
            (ContainsCurrent("America/Los_Angeles", ZonesThatUseLocationName) ||
            ContainsCurrent("Europe/Lisbon", ZonesThatUseLocationName) ||
            ContainsCurrent("Africa/Nairobi", ZonesThatUseLocationName));
    }

    // Some time zones may give better display names using their location names rather than their generic name.
    // We can update this list as need arises.
    private const string ZonesThatUseLocationName = "\n" +
        "Europe/Minsk\n" +      // Prefer "Belarus Time" over "Moscow Standard Time (Minsk)"
        "Europe/Moscow\n" +     // Prefer "Moscow Time" over "Moscow Standard Time"
        "Europe/Simferopol\n" + // Prefer "Simferopol Time" over "Moscow Standard Time (Simferopol)"
        "Pacific/Apia\n" +      // Prefer "Samoa Time" over "Apia Time"
        "Pacific/Pitcairn\n";   // Prefer "Pitcairn Islands Time" over "Pitcairn Time"

    private static bool ContainsCurrent(string pattern, string source)
    {
        int index = 1;
        do
        {
            index = source.IndexOf(pattern, index, StringComparison.OrdinalIgnoreCase);
            if (index < 0)
            {
                return false;
            }

            if (index + pattern.Length < source.Length && source[index - 1] == '\n' && source[index + pattern.Length] == '\n')
            {
                return true;
            }

            index += pattern.Length;
        } while (index < source.Length);

        return false;
    }
```

produces:

|     Method |      Mean |    Error |   StdDev | Gen 0 | Gen 1 | Gen 2 | Allocated |
|----------- |----------:|---------:|---------:|------:|------:|------:|----------:|
|  SimpleWay |  37.48 ns | 0.198 ns | 0.166 ns |     - |     - |     - |         - |
| CurrentWay | 556.16 ns | 3.266 ns | 3.055 ns |     - |     - |     - |         - |

>  code simplicity

I don't think the `DelimitedStringContains` method is simpler than looping over an array and checking for equality.

Reply:
I agree. my original suggestion was assuming we'll not concatenate the string we search for with the bound characters. so, I am fine going back using the array.

Reply:
Ok.  Will do.